sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -6326,6 +6346,7 @@ pi_result piEnqueueMemUnmap(pi_queue Queue, pi_mem MemObj, void *MappedPtr,
       }
     }
 
+    std::shared_lock Guard(MemObj->Mutex);
```

Comment:
The "memcpy" below potentially modifies the data of the MemObj, right?
```suggestion
    std::scoped_lock Guard(MemObj->Mutex);
```

Reply:
I agree, fixed.